File: 7. Reverse Integer Medium/solution.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(reverse(123))
	fmt.Println(reverse(-123))
	fmt.Println(reverse(120))
}

func reverse(x int) int {
	isNegative := x < 0
	x = int(math.Abs(float64(x)))
	reversedNumber := 0

	for x > 0 {
		digit := x % 10
		reversedNumber = reversedNumber*10 + digit
		x /= 10
	}

	if reversedNumber > math.MaxInt32 {
		return 0
	}

	if isNegative {
		return -reversedNumber
	}
	return reversedNumber
}

File: 7. Reverse Integer Medium/solution.py
def reverse(x: int) -> int:
    is_negative = x < 0
    x = abs(x)
    reversed_number = 0
    
    while x:
        digit = x % 10
        reversed_number = reversed_number * 10 + digit
        x //= 10
    
    if reversed_number > 2 ** 31 - 1:
        return 0
    
    return -reversed_number if is_negative else reversed_number

print(reverse(123))
print(reverse(-123))
print(reverse(120))